lib/services/blob/blobservice.core.js

```diff
@@ -1122,6 +1122,38 @@ BlobService.prototype.listBlobDirectoriesSegmentedWithPrefix = function (contain
+*
+* @this {BlobService}
+* @param {string}             container                           The container name.
+* @param {string}             prefix                              The prefix of the blob directory or blob name.
+* @param {object}             currentToken                        A continuation token returned by a previous listing operation. Please use 'null' or 'undefined' if this is the first operation.
+* @param {object}             [options]                           The request options.
+* @param {int}                [options.maxResults]                Specifies the maximum number of blobs or directories to return per call to Azure ServiceClient. This does NOT affect list size returned by this function. (maximum: 5000)
+* @param {LocationMode}       [options.locationMode]              Specifies the location mode used to decide which location the request should be sent to. 
+*                                                                 Please see StorageUtilities.LocationMode for the possible values.
+* @param {int}                [options.timeoutIntervalInMs]       The server timeout interval, in milliseconds, to use for the request.
+* @param {int}                [options.clientRequestTimeoutInMs]  The timeout of client requests, in milliseconds, to use for the request.
+* @param {int}                [options.maximumExecutionTimeInMs]  The maximum execution time, in milliseconds, across all potential retries, to use when making this request.
+*                                                                 The maximum execution time interval begins at the time that the client begins building the request. The maximum
+*                                                                 execution time is checked intermittently while performing requests, and before executing retries.
+* @param {string}             [options.clientRequestId]           A string that represents the client request ID with a 1KB character limit.
+* @param {bool}               [options.useNagleAlgorithm]         Determines whether the Nagle algorithm is used; true to use the Nagle algorithm; otherwise, false.
+*                                                                 The default value is false.
+* @param {errorOrResult}      callback                            `error` will contain information
+*                                                                 if an error occurs; otherwise `result` will contain `entries` and `continuationToken`. 
+*                                                                 `entries`  gives a list of `[directories]{@link BlobResult}` and the `continuationToken` is used for the next listing operation.
+*                                                                 `response` will contain information related to this operation.
+*/
+BlobService.prototype.listBlobsOrBlobDirectoriesSegmentedWithPrefix = function (container, prefix, currentToken, optionsOrCallback, callback) {
```

Comment:
Would you add test for this new API as well?

Reply:
Yes, but I need some help. I looked at the test suite and it's very complex, and not sure how to integrate a new test.